test/uk/gov/hmrc/helptosave/utils/MongoSupport.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.utils
+
+import java.net.ServerSocket
+
+import com.github.simplyscala.{MongoEmbedDatabase, MongodProps}
+import org.scalatest.Matchers
+import org.scalatest.concurrent.Eventually
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.FailoverStrategy
+import reactivemongo.core.actors.Exceptions.PrimaryUnavailableException
+import uk.gov.hmrc.mongo.MongoConnector
+
+trait MongoSupport extends MongoEmbedDatabase { this: Eventually with Matchers ⇒
+
+  def withMongo(f: ReactiveMongoComponent ⇒ Any): Unit = withMongo(mongo(_, None), f)
+
+  def withBrokenMongo(f: ReactiveMongoComponent ⇒ Any): Unit =
+    scala.util.control.Exception.ignoring(classOf[PrimaryUnavailableException]) {
+      withMongo(brokenMongo, f)
+    }
+
+  private def withMongo(newReactiveMongoComponent: MongodProps ⇒ ReactiveMongoComponent, f: ReactiveMongoComponent ⇒ Any): Unit = {
+    val port = {
+      val socket = new ServerSocket(0)
+      socket.close()
+
+      eventually {
+        socket.isClosed shouldBe true
+      }
+
+      socket.getLocalPort
+    }
```

Comment:
can you clarify why the above port logic is needed ?  you are reading `getLocalPort` from a closed socket, so not how it works

Reply:
is there any chance that some other process grabs this port before you pick it ?

Reply:
it's a small chance - something will have to grab the port in the milliseconds in between when we give up the port and mongo picking it up